test/AddressSpaceCast/issue-1096.ll

```diff
@@ -1,10 +1,10 @@
 ; RUN: clspv-opt %s -o %t.ll --passes=lower-addrspacecast
 ; RUN: FileCheck %s < %t.ll
 
-; CHECK: declare i32 @_Z8spirv.opjPU3AS3jjj(i32, ptr addrspace(3), i32, i32, i32)
+; CHECK-NOT: declare i32 @_Z8spirv.opjPU3AS3jjj(i32, ptr addrspace(3), i32, i32, i32)
```

Comment:
The goal of this line is to check that we converted the address space from `generic` (4) to `local` (3).
Please update the test so that this check is still performed.

Reply:
The change renames the "@_Z8spirv.opjPU3AS3jjj"  function to "@_Z8spirv.234.PU3AS3jjj". What is makes Replacing the name in the test is the correct fix?

Reply:
Yes, it would make sense.
But can you elaborate on why this change is needed? Do you have a cl source showing what the issue is? If yes, maybe it is worth adding a specific test in clspv.
Thanks

Reply:
The clvk/clspv project crashes when testing OpenCL-CTS
The test is: <CTS>/test_c11_atomics atomic_flag

The kernel is:
`__kernel void test_atomic_kernel(uint threadCount, uint numDestItems, __global int *finalDest, __global int *oldValues, volatile __local atomic_flag *destMemory)
{
  uint  tid = get_global_id(0);


      // initialize atomics not reachable from host (first thread
      // is doing this, other threads are waiting on barrier)
      if(get_local_id(0) == 0)
        for(uint dstItemIdx = 0; dstItemIdx < numDestItems; dstItemIdx++)
        {
          if(finalDest[dstItemIdx])
            atomic_flag_test_and_set_explicit(destMemory+dstItemIdx,
                                              memory_order_relaxed,
                                              memory_scope_work_group);
          else
            atomic_flag_clear_explicit(destMemory+dstItemIdx,
                                       memory_order_relaxed,
                                       memory_scope_work_group);    }
  barrier(CLK_LOCAL_MEM_FENCE);

  uint cnt, stop = 0;
  for(cnt = 0; !stop && cnt < threadCount; cnt++) // each thread must find critical section where it is the first visitor
  {
    bool set = atomic_flag_test_and_set_explicit(&destMemory[cnt], memory_order_relaxed, memory_scope_work_group);
    atomic_work_item_fence(CLK_LOCAL_MEM_FENCE | CLK_GLOBAL_MEM_FENCE, memory_order_acquire,memory_scope_work_group);
    if (!set)
    {
      uint csIndex = get_enqueued_local_size(0)*get_group_id(0)+cnt;
      // verify that thread is the first visitor
      if(oldValues[csIndex] == 1000000000)
      {
	oldValues[csIndex] = tid; // set the winner id for this critical section
	stop = 1;
      }
      atomic_work_item_fence(CLK_LOCAL_MEM_FENCE | CLK_GLOBAL_MEM_FENCE, memory_order_release,memory_scope_work_group);
      atomic_flag_clear_explicit(&destMemory[cnt], memory_order_relaxed, memory_scope_work_group);
    }
  }

  // Copy final values to host reachable buffer
  barrier(CLK_LOCAL_MEM_FENCE);
  if(get_local_id(0) == 0) // first thread in workgroup
    for(uint dstItemIdx = 0; dstItemIdx < numDestItems; dstItemIdx++)

        finalDest[dstItemIdx] =
            atomic_flag_test_and_set_explicit(destMemory+dstItemIdx,
                                              memory_order_relaxed,
                                              memory_scope_work_group);
}
`

Reply:
Got it, thank you 😀
Please update the test and that should be good to go for me